packages/gestalt/src/SideNavigationGroup.js

```diff
@@ -49,6 +53,25 @@ export type Props = {|
    *  When supplied, will display a notification dot. See the [Notification](https://gestalt.pinterest.systems/web/sidenavigation#Notification) variant to learn more.
    */
   notificationAccessibilityLabel?: string,
+   /**
+   * The primary action for each group. See the [primary action variant](https://gestalt.pinterest.systems/web/sidenavigation#Primary-action) to learn more.
+   */
+  primaryAction?: {|
+    icon?: 'ellipsis' | 'edit' | 'trash-can',
+    onClick?: ({|
+      event:
+        | SyntheticMouseEvent<HTMLDivElement>
+        | SyntheticKeyboardEvent<HTMLDivElement>
+        | SyntheticMouseEvent<HTMLAnchorElement>
+        | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    |}) => void,
+    tooltip: {|
+      accessibilityLabel?: string,
+      text: string,
+      zIndex?: Indexable,
+    |},
+    dropdownItems?: $ReadOnlyArray<Element<typeof Dropdown.Item>>,
```

Comment:
Do we allow Dropdown.Section or Dropdown.Link here?

Reply:
Mmm I'd keep it at just Dropdown.Item, as they are intended to be very small action dropdowns. We can always extend